src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -887,6 +884,15 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
         SetCreateStat(STAT_INTELLECT, 28);
     }
 
+    // Power
+    if (petType == HUNTER_PET) // Hunter pets have focus
+        SetPowerType(POWER_FOCUS);
+    else if (IsPetGhoul() || IsPetAbomination()) // DK pets have energy
+        SetPowerType(POWER_ENERGY);
+    else 
+        SetPowerType(POWER_MANA);
```

Comment:
pretty sure this one is wrong - all warlock pets use energy (http://www.wowhead.com/imp-abilities)

Reply:
Ok, this will be the only difference for now between 3.3.5a and 7.x. 3.3.5 could use this PR aswell, right?

Reply:
No idea about that